### ISSUE TITLE
create deep links for the C# specification.

### ISSUE THREAD

**Issue by BillWagner** [dotnet/docs#8668]
There are 78 files that have links to the official C# spec. All of the listed files link to the introduction of the spec. Linking to the relevant of the spec would be a much better experience.

The destination links should be in the form:

```md
[some topic](~/_csharplang/spec/some-file.md#anchor-for-the-section) 
```

The source for the C# 6.0 spec is in the [dotnet/csharplang](https://github.com/csharplang) repo under the "spec" directory. Every heading should have an anchor tag that you can click to see the correct anchor in the browser address bar.

The blank lines in the list below suggest a possible breakdown for reasonable PR sizes. Community members that don't have write access won't be able to check the boxes. If you are submitting a PR for this issue, tag the issue, and we'll update the check list when we review and merge the PR.  

If you find files that have already been updated with appropriate deep links (some have), add a comment here to check off that file.

- [ ] /docs/csharp/language-reference/keywords/abstract.md
- [ ] /docs/csharp/language-reference/keywords/accessibility-domain.md
- [ ] /docs/csharp/language-reference/keywords/accessibility-levels.md
- [x] /docs/csharp/language-reference/keywords/as.md
- [x] /docs/csharp/language-reference/keywords/byte.md
- [ ] /docs/csharp/language-reference/keywords/event.md
- [ ] /docs/csharp/language-reference/keywords/extern-alias.md
- [ ] /docs/csharp/language-reference/keywords/get.md
- [ ] /docs/csharp/language-reference/keywords/in-parameter-modifier.md
- [x] /docs/csharp/language-reference/keywords/int.md

---------------------

- [x] /docs/csharp/language-reference/keywords/internal.md
- [x] /docs/csharp/language-reference/keywords/long.md
- [x] /docs/csharp/language-reference/keywords/nameof.md
- [x] /docs/csharp/language-reference/keywords/sbyte.md
- [ ] /docs/csharp/language-reference/keywords/sealed.md
- [ ] /docs/csharp/language-reference/keywords/set.md
- [x] /docs/csharp/language-reference/keywords/short.md
- [ ] /docs/csharp/language-reference/keywords/sizeof.md
- [ ] /docs/csharp/language-reference/keywords/static.md
- [ ] /docs/csharp/language-reference/keywords/string.md

---------------------

- [ ] /docs/csharp/language-reference/keywords/this.md
- [ ] /docs/csharp/language-reference/keywords/throw.md
- [ ] /docs/csharp/language-reference/keywords/true-literal.md
- [ ] /docs/csharp/language-reference/keywords/true-operator.md
- [ ] /docs/csharp/language-reference/keywords/try-catch-finally.md
- [ ] /docs/csharp/language-reference/keywords/try-catch.md
- [ ] /docs/csharp/language-reference/keywords/try-finally.md
- [x] /docs/csharp/language-reference/keywords/typeof.md
- [x] /docs/csharp/language-reference/keywords/uint.md
- [x] /docs/csharp/language-reference/keywords/ulong.md

---------------------

- [ ] /docs/csharp/language-reference/keywords/unchecked.md
- [ ] /docs/csharp/language-reference/keywords/unsafe.md
- [x] /docs/csharp/language-reference/keywords/ushort.md
- [ ] /docs/csharp/language-reference/keywords/using-directive.md
- [ ] /docs/csharp/language-reference/keywords/using-statement.md
- [ ] /docs/csharp/language-reference/keywords/value.md
- [ ] /docs/csharp/language-reference/keywords/virtual.md
- [ ] /docs/csharp/language-reference/keywords/volatile.md
- [ ] /docs/csharp/language-reference/keywords/yield.md

---------------------

- [x] /docs/csharp/language-reference/operators/conditional-and-operator.md
- [ ] /docs/csharp/language-reference/operators/index-operator.md
- [ ] /docs/csharp/language-reference/operators/invocation-operator.md
- [ ] /docs/csharp/language-reference/operators/member-access-operator.md
- [ ] /docs/csharp/language-reference/operators/namespace-alias-qualifer.md
- [x] /docs/csharp/language-reference/operators/null-conditional-operators.md

---------------------

- [ ] /docs/csharp/programming-guide/arrays/index.md
- [ ] /docs/csharp/programming-guide/classes-and-structs/access-modifiers.md
- [ ] /docs/csharp/programming-guide/classes-and-structs/constants.md
- [ ] /docs/csharp/programming-guide/classes-and-structs/destructors.md
- [ ] /docs/csharp/programming-guide/classes-and-structs/fields.md
- [ ] /docs/csharp/programming-guide/classes-and-structs/index.md
- [ ] /docs/csharp/programming-guide/classes-and-structs/methods.md
- [ ] /docs/csharp/programming-guide/classes-and-structs/named-and-optional-arguments.md

---------------------

- [x] /docs/csharp/programming-guide/classes-and-structs/partial-classes-and-methods.md
- [x] /docs/csharp/programming-guide/classes-and-structs/passing-parameters.md
- [x] /docs/csharp/programming-guide/classes-and-structs/private-constructors.md
- [x] /docs/csharp/programming-guide/classes-and-structs/properties.md
- [x] /docs/csharp/programming-guide/classes-and-structs/static-classes-and-static-class-members.md
- [x] /docs/csharp/programming-guide/classes-and-structs/using-constructors.md

---------------------

- [x] /docs/csharp/programming-guide/delegates/index.md
- [x] /docs/csharp/programming-guide/events/index.md
- [x] /docs/csharp/programming-guide/exceptions/creating-and-throwing-exceptions.md
- [x] /docs/csharp/programming-guide/exceptions/exception-handling.md
- [x] /docs/csharp/programming-guide/exceptions/index.md
- [x] /docs/csharp/programming-guide/indexers/index.md
- [x] /docs/csharp/programming-guide/inside-a-program/general-structure-of-a-csharp-program.md
- [ ] /docs/csharp/programming-guide/inside-a-program/index.md
- [x] /docs/csharp/programming-guide/interop/index.md

---------------------

- [ ] /docs/csharp/programming-guide/statements-expressions-operators/anonymous-functions.md
- [ ] /docs/csharp/programming-guide/statements-expressions-operators/index.md
- [ ] /docs/csharp/programming-guide/statements-expressions-operators/lambda-expressions.md
- [ ] /docs/csharp/programming-guide/statements-expressions-operators/statements.md
- [ ] /docs/csharp/programming-guide/types/boxing-and-unboxing.md
- [ ] /docs/csharp/programming-guide/types/casting-and-type-conversions.md
- [ ] /docs/csharp/programming-guide/types/index.md
- [ ] /docs/csharp/programming-guide/unsafe-code-pointers/arithmetic-operations-on-pointers.md
- [ ] /docs/csharp/programming-guide/unsafe-code-pointers/index.md
- [ ] /docs/csharp/programming-guide/xmldoc/xml-documentation-comments.md

**Comment by TheSench** [comment]
In adding links for `internal`, I also added `public`, `private`, and `protected`; as I noticed they were missing deep-links.

**Comment by pkulikov** [comment]
@BillWagner, as not all topics in the list are yet updated, this issue was closed by accident.

**Comment by TheSench** [comment]
My apologies, I am new to Pull Requests and marked them all as "Fixes #8668" instead of "Contributes to #8668".  Bill caught most of them, but missed the last PR, which caused this issue to be closed.
